### PR TITLE
Add Windows docker container support for building Open Enclave in Jenkins

### PR DESCRIPTION
**Pull request by johnkord**

### CODE REVIEW

File: .jenkins/build_docker_images.Jenkinsfile [hunk 1]
```diff
@@ -7,9 +7,33 @@ OETOOLS_REPO = "https://oejenkinscidockerregistry.azurecr.io"
...
+                        puboefullWin2016.push('latest')
+                }
+            }
+        }
+    }
+}
+
+def buildLinuxDockerImages() {
+    stage("nonSGX") {
+        node('SGXFLC-Windows-DCAP') {
```
[the remark this answers is not in the document]
Comment (johnkord) [reply .jenkins/build_docker_images.Jenkinsfile]: Fixed :) Thanks! Had higher priorities than fixing this PR recently!

File: .jenkins/Dockerfile.full.WindowsServer [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+# escape=`
+FROM mcr.microsoft.com/windows/servercore:ltsc2016
+
+COPY scripts/install-windows-prereqs.ps1 /
+RUN powershell -c "/install-windows-prereqs.ps1 -InstallPath C:\openenclave -LaunchConfiguration SGX1FLC-NoDriver -DCAPClientType Azure"
```
Comment (andschwa) [inline .jenkins/Dockerfile.full.WindowsServer]: Coooool

**Comment by johnkord** [comment]
Yes, it works @andschwa  :) You can even expose the sgx device on Windows to a process-isolated Windows container *IF* you're using the SGX1FLC launch configuration, for launching enclaves & running tests in a Windows docker container!

**Comment by BRMcLaren** [comment]
Hey @johnkord,

Running these scripts locally, the image script hangs after installation and does not complete the build. 

I,e,

``Attempting to resolve dependencies for package 'EnclaveCommonAPI.1.0.100' with DependencyBehavior 'Lowest'
Resolving actions to install package 'EnclaveCommonAPI.1.0.100'
Resolved actions to install package 'EnclaveCommonAPI.1.0.100'
Adding package 'EnclaveCommonAPI.1.0.100' to folder 'C:\Downloads\prereqs\nuget'
Added package 'EnclaveCommonAPI.1.0.100' to folder 'C:\Downloads\prereqs\nuget'
Successfully installed 'EnclaveCommonAPI 1.0.100' to C:\Downloads\prereqs\nuget
[2019-10-10T11:57:15.3114077-07:00] Installing VC 2012 runtime
Please reboot your computer for the configuration to complete.
`

I wanted to double-check that this was not an issue for Jenkins?

**Comment by jhand2** [comment]
@BRMcLaren @johnkord Can this be merged or are changes needed? If ready to be merged, please add DCO to the commit.

**Comment by radhikaj** [comment]
@BRMcLaren  I am closing this PR as of now. Please reopen when it is ready to be merged.

**Pull request closed without merging**